docs/plans/2026-06-10-wave2-guard-precision-coverage.md

```
@@ -0,0 +1,182 @@
+/// final path component, minus SAFE_SUFFIXES. Strips one leading `@`
+/// (curl/httpie upload operands) and trailing `)` before classifying.
+pub fn is_dangerous_env_token(token: &str) -> bool
+```
+
+Baked-in decisions: `@.env` counts (curl exfil idiom); `.envrc` dangerous (preserves today's substring behavior); `settings.environment` / `.environment` / `my.envelope.txt` clean (kills the #86 substring class for both guards).
+
+**Redesign `bash_leaks_secrets` (:110-185).** Delete `read_operand` (:12-19) and the six-verb list (:115). New flow under the existing `contains(".env")` quick-reject: for each `command_segments` segment, `tokenize`; command word = basename of token 0; if on the metadata-safe allowlist → skip segment; else block if any subsequent token has no internal whitespace AND `is_dangerous_env_token`. The whitespace rule is the FP firewall: quoted prose stays glued via tokenize (`git commit -m "...env..."` → multi-word token → skipped); a quoted filename `".env"` stays a clean token → caught.
+
+**Metadata-safe allowlist** (never emit file contents): `ls, stat, file, du, wc, find, touch, mkdir, chmod, chown, rm, echo, printf, basename, dirname, realpath, test, [, direnv, git`. Notables: `rm` allowed here because prevent-secret-writes already blocks `rm .env` with the right message (avoids double block, wrong rationale); `git` allowed (`git add .env` is staging, not a context leak; `.env` is gitignored in practice — residual `git show <ref>:.env` gap doc-commented); `direnv allow .envrc` is the sanctioned workflow the Fix line recommends. `cp, mv, ln, tar` deliberately NOT listed — `cp .env /tmp/leak` is filesystem exfil and blocks.
+
+`source` / `. ` branches (:133-151) collapse into the generic rule (not safe-listed → blocked); dot-source FP protection becomes structural (in `grep . .env`, `.` is an argument not a command word).
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)